src/AppBundle/Service/EventProcessor.php

```diff
@@ -535,18 +538,17 @@ private function setUserCounts(EventWiki $wiki, EventWikiRepository $ewRepo): vo
      */
     private function getParticipantNames(): array
     {
-        // Quick cache.
-        static $parUsernames = null;
-        if (null !== $parUsernames) {
-            return $parUsernames;
```

Comment:
FYI @MaxSem , this was causing erroneous results for https://github.com/wikimedia/eventmetrics/pull/217/files#diff-421e5b20693d341d563a6dec2f088b95R441. Looking at that test, indeed the new editor and retention metrics should have been zero. The cheap "quick cache" code above was making it pull in the usernames from the previous test. My fault.

Reply:
To be clear, I wrote the bad caching code, you wrote the test which was asserting the wrong result. Definitely my fault, just thought I'd share this discovery.